include/cctz/string_detail.h

```diff
@@ -0,0 +1,90 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+#ifndef CCTZ_CIVIL_STRING_DETAIL_H_
+#define CCTZ_CIVIL_STRING_DETAIL_H_
+
+#include <cctype>
+#include <cstring>
+#include <string>
+
+namespace cctz {
+namespace detail {
+
+struct char_range {
```

Comment:
It is strange for `string_detail.h` to declare `char_range`.  Why not `char_range.h`?

But that said, why not implement `cctz::string_view` with whatever subset of the `std::string_view` interface you need, so that we can simply move to the latter if/when possible?

Reply:
frankly, I sort of arrived at this incrementally, so it wasn't deeply thought out.  I had intended to make something with a much smaller interface than string_view so it didn't have as much baggage to go with it.

Would you like me to change to something like cctz::string_view?

Reply:
I think so, yes.  There is no need to implement the entire interface though ... just the subset you need to get the job at hand done (i.e., nothing more than you already have).

Reply:
Greg mentioned that he wasn't sure what I had said here, so allow me to restate more clearly.

Add `include/cctz/string_view.h`, which implements `cctz::string_view` with a strict subset of the `std::string_view` interface.  Then use `cctz::string_view` in whatever interfaces you chose, safe in the knowledge that we'll be able to switch to `std::string_view` in the future.

I'm even happy to do that if you like.

Reply:
if you are happy to take it over, that would be great.  I tried to do this for a few hours and discovered that the existing parser traffics in `const char*` very very hard, so migrating it to something that does not expose `const char*` public data members is quite awkward